### PR TITLE
create ZIO from R => Future[A]

### PR DESCRIPTION
**Pull request by jczuchnowski**
A method that allows to lift a function returning a `Future` into `ZIO` that requires the input to the function as its environment.
Recently had a case where this would come in handy. I'm not sure about the naming though.

### CODE REVIEW

File: core/shared/src/main/scala/zio/ZIO.scala [hunk 1]
```diff
@@ -2140,6 +2140,12 @@ private[zio] trait ZIOFunctions extends Serializable {
   final def fromFunctionM[R, E, A](f: R => IO[E, A]): ZIO[R, E, A] =
     environment[R].flatMap(f)
 
+  /**
+   * Lifts a function returning Future into an effect that requires the input to the function.
+   */
+  final def fromFunctionF[R, A](f: R => scala.concurrent.Future[A]): RIO[R, A] =
```
Comment (jdegoes) [inline core/shared/src/main/scala/zio/ZIO.scala]: `fromFunctionFuture`?

Comment (jczuchnowski) [reply core/shared/src/main/scala/zio/ZIO.scala]: I thought about it originally. Seemed too long, but if you thought about the same then maybe it's actually a good name ;)

**Comment by ioleo** [comment]
Nice one :+1:

**Comment by jdegoes** [comment]
@jczuchnowski Please re-open this when you are ready to continue! Just needs alphabetical ordering and additions to the companion objects of the type aliases.

**Comment by jczuchnowski** [comment]
Sure, just didn't have time to get to it. But the hackathon is coming... :smile:

**Comment by jczuchnowski** [comment]
@jdegoes Sure, just didn't have time to get to it. But the hackathon is coming... :smile:

**Comment by jczuchnowski** [comment]
Ah, who needs sleep anyway... :smile:

**Comment by jczuchnowski** [comment]
@jdegoes @iravid looks like a failing flaky test. What's the procedure here?

**Comment by jczuchnowski** [comment]
@jdegoes looks like it's ready to merge now

**Pull request merged**